internal/cmd/signer.go

```diff
@@ -13,9 +13,21 @@ import (
 
 // GetSigner returns a signer according to the CLI context.
 func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
+	// Check if using on-demand key
+	if opts.KeyID != "" && opts.PluginName != "" && opts.Key == "" {
+		// Construct a signer from on-demand key
+		mgr := plugin.NewCLIManager(dir.PluginFS())
+		plugin, err := mgr.Get(context.Background(), opts.PluginName)
```

Comment:
We should use the context generated in `runSign()` because it contains the logger.

Reply:
Also please update line 41 to use the context from `runSign()`

Reply:
Added ctx (the context from runSign) to the function in latest commit